MSET9_installer_script/mset9.py

```diff
@@ -572,344 +637,411 @@ def writeProtectCheck():
+trigger = "002F003A.txt"  # all 3ds ":/" in hex format
+triggerFilePath = ""
 
-# make a table so we can print regions based on what hex code from the above is found
-regionTable = {
-	0x8F: "USA Region",
-	0x98: "EUR Region",
-	0x82: "JPN Region",
-	0xA1: "CHN Region",
-	0xA9: "KOR Region",
-	0xB1: "TWN Region"
-}				
-
-homeDataPath, miiDataPath, homeHex, miiHex = "", "", 0x0, 0x0
-def sanity():
-	global fs, haxState, realId1Path, id0, id1, homeDataPath, miiDataPath, homeHex, miiHex
-	menuExtdataGood = False
-	miiExtdataGood = False
+def createHaxID1():
+	global fs, ID0, hackedID1Path, realID1Path, realID1BackupTag
 
+	print("\033[0;33m=== DISCLAIMER ===\033[0m") # 5;33m? The blinking is awesome but I also don't want to frighten users lol
+	print()
+	print("This process will temporarily reset all your 3DS data.")
+	print("All your applications and themes will disappear.")
+	print("This is perfectly normal, and if everything goes right, it will re-appear")
+	print("at the end of the process.")
+	print()
+	print("In any case, it is highly recommended to make a backup of your SD card's contents to a folder on your PC.")
+	print("(Especially the 'Nintendo 3DS' folder.)")
 	print()
-	prinfo("Performing sanity checks...")
-
-	writeProtectCheck()
-
-	prinfo("Ensuring extracted files exist...")
-	fileSanity = 0
-	fileSanity += softcheck("boot9strap/boot9strap.firm", 0, 0x08129C1F, 1)
-	fileSanity += softcheck("boot.firm", retval = 1)
-	fileSanity += softcheck("boot.3dsx", retval = 1)
-	fileSanity += softcheck("b9", retval = 1)
-	fileSanity += softcheck("SafeB9S.bin", retval = 1)
-	if fileSanity > 0:
-		prbad("Error 08: One or more files are missing or malformed!")
-		prinfo("Please re-extract the MSET9 zip file, overwriting any existing files when prompted.")
-		exitOnEnter()
-	prgood("All files look good!")
 
-	prinfo("Checking databases...")
-	checkTitledb = softcheck(realId1Path + "/dbs/title.db", 0x31E400, 0, 1)
-	checkImportdb = softcheck(realId1Path + "/dbs/import.db", 0x31E400, 0, 1)
-	if checkTitledb or checkImportdb:
-		prbad("Error 10: Database(s) malformed or missing!")
-		if not (
-			fs.exists(realId1Path + "/dbs/import.db")
-			or fs.exists(realId1Path + "/dbs/title.db")
-		):
-			if not fs.exists(realId1Path + "/dbs"):
-				fs.mkdir(realId1Path + "/dbs")
-			if checkTitledb:
-				fs.open(realId1Path + "/dbs/title.db", "x").close()
-			if checkImportdb:
-				fs.open(realId1Path + "/dbs/import.db", "x").close()
-
-			prinfo("Created empty databases.")
-		prinfo("Please initialize the title database by inserting the SD into your console, powering it on, then navigating to System Settings -> Data Management -> Nintendo 3DS -> Software -> Reset, then rerun this script.")
-		prinfo("Visual guide: https://3ds.hacks.guide/images/screenshots/database-reset.jpg")
-		exitOnEnter()
+	if osver == "Linux": # ...
+		print("(on Linux, things like to not go right - please ensure that your SD card is mounted with the 'utf8' option.)")
```

Comment:
Second line of information can help a person, my linux mounted automatically utf-8 but that may be of my system locale pt_PT.UTF-8, otherwise can tell the user to mount again with the argument `utf8`, `utf8=1`, `utf8=yes` or `utf8=true` and `uni_xlate` argument must not be set, `iocharset=utf8` also discouraged for vfat.
Using `mount -o remount,utf8` does not fully give intended result for vfat remount as utf8!
We can also test if we are likely ready for utf-8, `open()` with `wb` will fail with OSError exception over errno 22 with characters invalid for vfat if it isn't mounted with utf-8, something we can test in try and except block, OSError exception has the attribute `.errno` with the integer to check.